presto-main/src/main/java/io/prestosql/metadata/TableHandle.java

```diff
@@ -17,23 +17,31 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import io.prestosql.connector.ConnectorId;
 import io.prestosql.spi.connector.ConnectorTableHandle;
+import io.prestosql.spi.connector.ConnectorTableLayoutHandle;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
 
-import java.util.Objects;
+import java.util.Optional;
 
 import static java.util.Objects.requireNonNull;
 
 public final class TableHandle
 {
     private final ConnectorId connectorId;
     private final ConnectorTableHandle connectorHandle;
+    private final ConnectorTransactionHandle transaction;
+    private final Optional<ConnectorTableLayoutHandle> layout;
```

Comment:
Could you add a Javadoc? E.g: that it's used only for table scans and it's temporary. Maybe add a TODO too